Content.Server/Health/BodySystem/Surgery/SurgeryData/BiologicalSurgeryData.cs

```diff
@@ -11,81 +12,125 @@
-                            return CautizerizeIncisionSurgery;
-                    }
-                    else
-                    {
-                        if (toolType == SurgeryToolType.Retraction)
-                            return RetractSkinSurgery;
-                        else if (toolType == SurgeryToolType.Cauterization)
-                            return CautizerizeIncisionSurgery;
-                    }
-                }
-                else
-                {
-                    if (toolType == SurgeryToolType.VesselCompression)
-                        return ClampVesselsSurgery;
-                    else if (toolType == SurgeryToolType.Cauterization)
-                        return CautizerizeIncisionSurgery;
-                }
+                return RemoveBodyPartSurgery;
             }
-            else
+            if (!_skinOpened) //Case: skin is normal.
             {
-                if (toolType == SurgeryToolType.Incision)
+                if (toolType == SurgeryType.Incision)
                     return OpenSkinSurgery;
             }
+            else if (_skinOpened && !_vesselsClamped) //Case: skin is opened, but not clamped.
+            {
+                if (toolType == SurgeryType.VesselCompression)
+                    return ClampVesselsSurgery;
+                else if (toolType == SurgeryType.Cauterization)
+                    return CautizerizeIncisionSurgery;
+            }
+            else if (_skinOpened && _vesselsClamped && !_skinRetracted) //Case: skin is opened and clamped, but not retracted.
+            {
+                if (toolType == SurgeryType.Retraction)
+                    return RetractSkinSurgery;
+                else if (toolType == SurgeryType.Cauterization)
+                    return CautizerizeIncisionSurgery;
+            }
+            else if (_skinOpened && _vesselsClamped && _skinRetracted) //Case: skin is fully open.
+            {
+                if (_parent.Mechanisms.Count > 0 && toolType == SurgeryType.VesselCompression && (_disconnectedOrgans.Except(_parent.Mechanisms).Count() != 0 || _parent.Mechanisms.Except(_disconnectedOrgans).Count() != 0))
+                    return LoosenOrganSurgery;
+                else if (_disconnectedOrgans.Count > 0 && toolType == SurgeryType.Incision)
+                    return RemoveOrganSurgery;
+                else if (toolType == SurgeryType.Cauterization)
+                    return CautizerizeIncisionSurgery;
+            }
             return null;
         }
 
-        protected void OpenSkinSurgery(BodyManagerComponent target, IEntity performer)
+        public override string GetDescription()
+        {
+            string toReturn = "";
+            if (_skinOpened && !_vesselsClamped) //Case: skin is opened, but not clamped.
+            {
+                toReturn += "The skin on his " + _parent.Name + " has an incision, but it is prone to bleeding.\n";
```

Comment:
Should be using string interpolation, localisation, and gender formatters.

Reply:
the what now

Reply:
```suggestion
                toReturn += Loc.GetString($"The skin on ${0:Their} " + _parent.Name + " has an incision, but it is prone to bleeding.\n", Entity);
```

You'll need to get the entity from somewhere, tho.

Reply:
Fixed after talking on discord